whylog/config/rule.py

```diff
@@ -19,10 +19,10 @@ def serialize(self):
             "constraints": self._constraints,
         }
 
-    def get_new_parsers(self, old_parsers):
+    def get_new_parsers(self, parser_name_generator):
         new_parsers = []
         for parser in itertools.chain([self._effect], self._causes):
-            if parser.name not in old_parsers:
+            if parser_name_generator.is_free_parser_name(parser.name):
```

Comment:
I am not sure if it works in web version when we have a few Teachers and every of them has its own blacklist. 
Look at the case:
- We have 2 Teachers: A and B.
- A wants a name for parser P1 "Error occurred on (.*)".
- B wants a name for parser P2 "Error occured on (.*) while data recovery".
- Config has empty praser names base.

Steps are:
1. A asks Config for name. He gets "error_occured"
2. B asks Config for name. He also gets "error_occured"
3. A saves rule, so name "error_occured" is inserted into Config base.
4. B saves rule. P2 is not saved because its name is in Config base and user even doesn't know about it!

Do I understand it right?